ec2/al2-managed.yaml

```diff
@@ -0,0 +1,846 @@
+    Default: UTC
+  BackupRetentionPeriod:
+    Description: 'The number of days to keep backups of the EC2 instance (set to 0 to disable).'
+    Type: Number
+    MinValue: 0
+    MaxValue: 35
+    Default: 30
+  BackupScheduleExpression:
+    Description: 'A CRON expression specifying when AWS Backup initiates a backup job.'
+    Type: String
+    Default: 'cron(0 4 ? * * *)'
+  RestoreImageId:
+    Description: 'Restore from existing Amazon Machine Image (AMI).'
+    Type: String
+    Default: ''
+  AmazonLinux2Version:
+    Description: 'Version of Amazon Linux 2 to boot from the first time (ignored if RestoreImageId is set; select the latest version for new stacks; OS updates are enabled to patch the system).'
+    Type: String
+    Default: '20191217.0'
+    AllowedValues: ['20191217.0']
+Mappings:
+  VersionMap:
+    '20191217.0':
+      Map: RegionMap201912170
+  RegionMap201912170:
+    'eu-north-1':
+      AMI: 'ami-0662eb9b9b8685935'
+    'ap-south-1':
+      AMI: 'ami-0217a85e28e625474'
+    'eu-west-3':
+      AMI: 'ami-007fae589fdf6e955'
+    'eu-west-2':
+      AMI: 'ami-0089b31e09ac3fffc'
+    'eu-west-1':
+      AMI: 'ami-0713f98de93617bb4'
+    'ap-northeast-2':
+      AMI: 'ami-0bea7fd38fabe821a'
+    'me-south-1':
+      AMI: 'ami-05f93aaf03b9bf20c'
+    'ap-northeast-1':
+      AMI: 'ami-011facbea5ec0363b'
+    'sa-east-1':
+      AMI: 'ami-09de7b4017733e2af'
+    'ca-central-1':
+      AMI: 'ami-0a269ca7cc3e3beff'
+    'ap-east-1':
+      AMI: 'ami-db3d78aa'
+    'ap-southeast-1':
+      AMI: 'ami-05c64f7b4062b0a21'
+    'ap-southeast-2':
+      AMI: 'ami-0b8b10b5bf11f3a22'
+    'eu-central-1':
+      AMI: 'ami-07cda0db070313c52'
+    'us-east-1':
+      AMI: 'ami-062f7200baf2fa504'
+    'us-east-2':
+      AMI: 'ami-02ccb28830b645a41'
+    'us-west-1':
+      AMI: 'ami-03caa3f860895f82e'
+    'us-west-2':
+      AMI: 'ami-04590e7389a6e577c'
+Conditions:
+  HasKmsKey: !Not [!Equals [!Ref ParentKmsKeyStack, '']]
+  HasKeyName: !Not [!Equals [!Ref KeyName, '']]
+  HasIAMUserSSHAccess: !Equals [!Ref IAMUserSSHAccess, 'true']
+  HasSSHBastionSecurityGroup: !Not [!Equals [!Ref ParentSSHBastionStack, '']]
+  HasNotSSHBastionSecurityGroup: !Equals [!Ref ParentSSHBastionStack, '']
+  HasAlertTopic: !Not [!Equals [!Ref ParentAlertStack, '']]
+  HasZone: !Not [!Equals [!Ref ParentZoneStack, '']]
+  HasIngressTcpPort1: !Not [!Equals [!Ref IngressTcpPort1, '']]
+  HasIngressTcpPort2: !Not [!Equals [!Ref IngressTcpPort2, '']]
+  HasIngressTcpPort3: !Not [!Equals [!Ref IngressTcpPort3, '']]
+  HasClientSecurityGroup1: !Not [!Equals [!Ref ParentClientStack1, '']]
+  HasClientSecurityGroup2: !Not [!Equals [!Ref ParentClientStack2, '']]
+  HasClientSecurityGroup3: !Not [!Equals [!Ref ParentClientStack3, '']]
+  HasManagedPolicyArns: !Not [!Equals [!Ref ManagedPolicyArns, '']]
+  HasBackupRetentionPeriod: !Not [!Equals [!Ref BackupRetentionPeriod, 0]]
+  HasRestoreImageId: !Not [!Equals [!Ref RestoreImageId, '']]
+Resources:
+  RecordSet:
```

Comment:
What about use cases in private subnets?

Reply:
Private Subnets are somewhat complicated. Proper support would require a new template. Because the EC2 instance resource depends on the EIP association which is not really needed in private subnets. So we could drop private subnets in this template altogether?